File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        set_nums=set()
        length=len(nums)
        l,r=0,0
        while r<length:
            n=nums[r]
            if r-l<=k:
                if n in set_nums:
                    return True
                set_nums.add(n)
                r+=1
            else:
                set_nums.discard(nums[l])
                l+=1
        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
       // If we use Priority queue the time complexity goes to O(nlogn)
       //Bucket sort time O(N) and space O(N)
       HashMap<Integer,Integer> frequency = new HashMap<>();
       for(int n:nums){
            frequency.put(n,frequency.getOrDefault(n,0)+1);
       }
       int n=nums.length;
       //int[] bucket = new int[];
       List<Integer>[] bucket = new List[n+1];
       for(int i=0;i<=n;i++){
            bucket[i]= new ArrayList<>();
    }
       for(int key:frequency.keySet()){
            //List<Integer> list = bucket[frequency.get(key)]
            bucket[frequency.get(key)].add(key);
       }
       List<Integer> ans = new ArrayList<>();
       int j=n;
       //int i=k;
       while(k>0){
            List<Integer> l=bucket[j];
            j--;
            if(l.size()==0){
                continue; 
                }
            ans.addAll(l);
            k=k-l.size();
       }
    return ans.stream().mapToInt(Integer::intValue).toArray();
    }
}